frontend/src/metabase/dashboard/containers/DashboardApp/DashboardApp.tsx

```diff
@@ -171,24 +188,52 @@ const DashboardApp = (props: DashboardAppProps) => {
     onTimeout,
   });
 
+  const { refreshDashboard } = useRefreshDashboard({
+    dashboardId: dashboardId,
+    parameterQueryParams,
+  });
+
   const {
-    documentTitle: _documentTitle,
-    pageFavicon: _pageFavicon,
-    isRunning: _isRunning,
-    isLoadingComplete: _isLoadingComplete,
-    children,
-    ...dashboardProps
-  } = props;
+    hasNightModeToggle,
+    hideDownloadButton,
+    hideParameters,
+    isFullscreen,
+    isNightMode,
+    onNightModeChange,
+    refreshPeriod,
+    onFullscreenChange,
+    setRefreshElapsedHook,
+    onRefreshPeriodChange,
+    theme,
+    titled,
+    font,
+  } = useDashboardUrlParams({ location, onRefresh: refreshDashboard });
+
+  useSyncURLSlug({ location });
 
   return (
     <div className={cx(CS.shrinkBelowContentSize, CS.fullHeight)}>
       <LeaveConfirmationModal isEnabled={isEditing && isDirty} route={route} />
-
-      {/* @ts-expect-error for now until we can get the prop-drilled types sorted out. Previously DashboardControls was a JS file so types weren't checked, but now there's a Pandora's box here */}
+      {/* @ts-expect-error - types coming from thunks are not correct*/}
       <Dashboard
-        dashboardId={getDashboardId(props)}
+        location={location}
+        dashboardId={dashboardId}
         editingOnLoad={editingOnLoad}
         addCardOnLoad={addCardOnLoad}
+        isFullscreen={isFullscreen}
+        refreshPeriod={refreshPeriod}
+        hideParameters={hideParameters}
+        isNightMode={isNightMode}
+        hasNightModeToggle={hasNightModeToggle}
+        setRefreshElapsedHook={setRefreshElapsedHook}
+        onNightModeChange={onNightModeChange}
+        onFullscreenChange={onFullscreenChange}
+        onRefreshPeriodChange={onRefreshPeriodChange}
+        hideDownloadButton={hideDownloadButton}
+        theme={theme}
```

Comment:
Can you double-check some of the passed props here? It's kind of out of scope, since refactor should be 1:1, but some of these props aren't even used in `Dashboard` e.g. if you look at `metabase/frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx`
- theme
- font
- titled
- hideDownloadButton

There could be more, but I'm too lazy to check 😄